src/hex.cpp

```diff
@@ -0,0 +1,86 @@
+#include <hex.hpp>
+#include <vector>
+#include <iostream>
+
+using namespace std;
+
+HexFormater::HexFormater(){
+  block_size = 1;
+  set_use_capital_letters(true);
+}
+
+int HexFormater::get_block_size(){
+  return block_size;
+}
+
+void HexFormater::set_block_size(int block_size){
+  this->block_size = block_size;
+}
+
+bool HexFormater::get_use_capital_letters(){
+  return use_capital_letters;
+}
+
+void HexFormater::set_use_capital_letters(bool use_capital_letters){
+  this->use_capital_letters = use_capital_letters;
+  offset = 7;
+  if(!use_capital_letters)
+    offset += 32;
+}
+
+string HexFormater::to_hex(byte data){
+  string s = "";
+  unsigned char tmp = (unsigned char) (data >> 4) % 16;
+
+  if (tmp > 9)
+    tmp += offset;
+  tmp += 48;
+  s += tmp;
+
+  tmp = (unsigned char)data % 16;
+
+  if (tmp > 9)
+    tmp += offset;
+  tmp += 48;
+  s += tmp;
+
+  return s;
+}
+
+string HexFormater::to_hex(BinaryBuffer *data){
+  string s = "";
+  vector<byte> *buffer = data->bytes();
+  for(unsigned int i=0; i<buffer->size(); i++){
+    s += to_hex(buffer->at(i));
+    if (i%1==0)
+      s+= " ";
+  }
+
+  return s;
+}
+
+byte HexFormater::hex_to_byte(string hex){
+  vector<char> data(hex.begin(), hex.end());
+  byte tmp, b;
+
+  if(data.size() < 2){
```

Comment:
Check if String is exactly two digits? Otherwise it will not fit in a byte anyways?
```suggestion
  if(data.size() != 2){
```